modules/tinymce/src/core/main/ts/api/UndoManager.ts

```diff
@@ -50,6 +51,32 @@ const UndoManager = (editor: Editor): UndoManager => {
       return Rtc.addUndoLevel(editor, undoManager, index, locks, beforeBookmark, level, event);
     },
 
+    /**
+     * Dispatch a change event if the content in last layer of the undoManager and the current editor content are different
+     *
+     * @method dispatchIfChanged
+     */
+    dispatchIfChanged: (): void => {
+      const data = undoManager.data;
+      const currentLevel = Levels.createFromEditor(editor);
+
+      const dispatchEvent = (lastLevel: UndoLevel) => {
+        editor.setDirty(true);
+        editor.dispatch('change', {
+          level: currentLevel,
+          lastLevel
+        });
+      };
+
+      Arr.get(data, index.get()).fold(
+        () => dispatchEvent(currentLevel),
```

Comment:
This implementation is incorrect. When there is no previous undo level the `lastLevel` should be `undefined` instead. You'll see that's what the old logic did (technically `null` though that was incorrect) and also what `Operations.ts` does as well.

Reply:
fixed, sorry